File: python/singular_to_plural.py
import re
import sys

word = input('Please enter the singular word:').strip()

if re.match(r'[s|sh|ch|o|x]$', word):
    print(word + 'es')
    sys.exit(0)

m = re.match(r'(.*)(fe?)', word)
if m:
    print(m.group(1) + 'ves')
    sys.exit(0)

m = re.match(r'(.*[^aiueo])y$', word)
if m:
    print(m.group(1) + 'ies')
    sys.exit(0)

print(word + 's')

File: rust/singular_to_plural.rs
use regex::Regex;

fn main() {
	println!("Please enter the singular word:")

	var w string
	fmt.Scan(&w)

	m, e := regexp.MatchString("[s|sh|ch|o|x]$", w)
	if e != nil {
		std::process::exit(1)
	}
	if m {
		println!(w + "es")
		os.Exit(0)
	}

	r, e := regexp.Compile("(.*)(fe?)$")
	if e != nil {
		std::process::exit(1)
	}
	f := r.FindStringSubmatch(w)
	if f != nil {
		println!(f[1] + "ves")
		os.Exit(0)
	}

	r2, e := regexp.Compile("(.*[^aiueo])y$")
	if e != nil {
		std::process::exit(1)
	}
	f2 := r2.FindStringSubmatch(w)
	if f2 != nil {
		println!(f2[1] + "ies")
		os.Exit(0)
	}

	println!(w + "s")
}
